### ISSUE TITLE
Getting 404 page not found error while accessing the data

### ISSUE THREAD

**Issue by agowa338** [ipfs/go-ipfs#7398]
I also run into this issue and it looked like a regression at first (as it occurred after an update).
I've
```
    "API": "/ip6/::1/tcp/5001",
    "Gateway": "/ip6/::1/tcp/8080"
```
in my configuration file and that results in the same error.

But if I change the API to the default of `/ip4/127.0.0.1/tcp/5001` it starts working again.
Using developer tools in the browser reveals what is going wrong in the first case:
![image](https://user-images.githubusercontent.com/2544867/71648704-f6642280-2d07-11ea-88aa-497cc18b7ada.png)

This can also be reproduced by using anything except the default like `/ip4/127.174.134.32/tcp/5001`.

To solve this not only the Gateway.HTTPHeaders section in the config file needs to be configured, but also the API.HTTPHeaders section. copy and pasting the section should be sufficient.

_Originally posted by @agowa338 in https://github.com/ipfs/go-ipfs/issues/6682#issuecomment-570100635_

**Comment by agowa338** [comment]
> @agowa338 can you open a new bug? The original issue is an issue with the Gateway. Your issue looks like a bug in the webui.

_Originally posted by @Stebalien in https://github.com/ipfs/go-ipfs/issues/6682#issuecomment-570261609_

**Comment by Stebalien** [comment]
Which version?

**Comment by agowa338** [comment]
It was the latest as time of writing.
I just checked again with ipfs desktop 0.11.4

The error seam to be partially fixed it is now working with firefox and chrome, but fails with edge (not the chromium one, the other one)...

Edge:
![image](https://user-images.githubusercontent.com/2544867/83585344-e916c280-a549-11ea-8b89-ae6134c8dac7.png)

The 404s are caused by the localization e.g. the "en-US" pages not being found, than a fallback occurs to "en" and everything is fine in chrome. But for edge it falls back to "de-DE" (also returns a HTTP-404 again and than it falls back to "de" and gets the resource with HTTP-200, but that's the only difference I could stop)...

Also the content of these HTTP-404's looks wird. It looks like a command is returned to the browser instead of being executed, here an example of such a response:
```
Content-Length: 197
Content-Type: text/plain; charset=utf-8
Date: Wed, 03 Jun 2020 01:23:02 GMT
X-Content-Type-Options: nosniff

ipfs resolve -r /ipfs/bafybeidatpz2hli6fgu3zul5woi27ujesdf5o5a7bu622qj6ugharciwjq/locales/de-DE/status.json: no link named "de-DE" under bafybeigx7h4jpkvd6q3n3qsqep6qrliwhux5i7frgr4qnsee5q5lpprsa4

```
This is the corresponding request:
```
Accept: */*
Accept-Encoding: gzip, deflate
Accept-Language: de-DE
Connection: Keep-Alive
DNT: 1
Host: [::1]:5001
Referer: http://[::1]:5001/ipfs/bafybeidatpz2hli6fgu3zul5woi27ujesdf5o5a7bu622qj6ugharciwjq/
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.102 Safari/537.36 Edge/18.19631
X-Requested-With: XMLHttpRequest
```

Tl;Dr: There are two issues, one in the js implementation that it fails to connect to the API if it is IPv6 only in *some* browsers. And the 2nd one that necessary HTTP-404's are thrown depending on the used locals.

I think there is no need to fix the "Edge"-case as it will be replaced by the chromium one anyway, it may be good to add it to the wiki though.
The other thing is the HTTP-404's, can we do something against those? They are quite irritating when debugging...

**Comment by Stebalien** [comment]
Are the 404s preventing the webui from _working_ or are they just network failures? It looks like we should be checking to see if we support a locale before we actually try to load it.

cc @lidel

> Also the content of these HTTP-404's looks wird. It looks like a command is returned to the browser instead of being executed, here an example of such a response:

That's normal, it's just a funky error message.

**Comment by lidel** [comment]
ipfs-webui checks if exact locale match exists, if not it retries with less specific ones until it finds a match with translated label, for example:
- `en-US` → `en`
- `en-GB` → `en`
- `zh-Hans` → `zh-CN` → `en`

We don't have translations for every possible language+country combination, that is why you see those "Not Found" errors – but that is normal and expected, part of translation logic. 

@agowa338 apart from those HTTP 404, is there any other error? 
If not, this issue can be closed.

**Comment by agowa338** [comment]
I don't know if the "IPv6 and Edge" one needs to be addressed, it only happens on the "pre installed Edge" but not on the latest one (because that is chromium based).

So thanks for the explanation of these 404s and I'm going to close this than.
